saleor/checkout/fetch.py

```diff
@@ -56,6 +54,66 @@ def get_country(self) -> str:
     def get_customer_email(self) -> str:
         return self.user.email if self.user else self.checkout.email
 
+    @property
+    def valid_shipping_methods(self) -> List["ShippingMethodData"]:
+        return [method for method in self.all_shipping_methods if method.active]
+
+
+@dataclass(frozen=True)
+class DeliveryMethodBase:
+    delivery_method: Optional["ShippingMethodData"] = None
+    shipping_address: Optional["Address"] = None
+
+    @property
+    def warehouse_pk(self) -> Optional[str]:
+        pass
+
+    @property
+    def delivery_method_order_field(self) -> dict:
+        return {"shipping_method": self.delivery_method}
+
+    @property
+    def is_local_collection_point(self) -> bool:
+        return False
+
+    @property
+    def delivery_method_name(self) -> Dict[str, Optional[str]]:
+        return {"shipping_method_name": None}
+
+    def get_warehouse_filter_lookup(self) -> Dict[str, Any]:
+        return {}
+
+    def is_valid_delivery_method(self) -> bool:
+        return False
+
+    def is_method_in_valid_methods(self, checkout_info: "CheckoutInfo") -> bool:
+        return False
+
+
+@dataclass(frozen=True)
+class ShippingMethodInfo(DeliveryMethodBase):
```

Comment:
We need the base as it's used in the business logic:
```
    if delivery_method is None:
        return DeliveryMethodBase()
```

I agree that we could drop some unnecessary attributes here

Reply:
My intention was do do the most of braking changes in the `CheckoutInfo` interface in 3.0. We are doing "some" at least, i.e. dropping the channel listing and changing types from `ShippingMethod` to `ShippingMethodData`. Having a delivery method base in 3.0 (just for shipping methods, without collection points) would result in less conflict overall.

Reply:
While "deliveries" are a concept from 3.1, I think that it makes sense to start using this name in 3.0 - it will make the codebase more similar to what we have in 3.1 which is a good thing.